File: src/main/java/io/github/chaosdave34/kitpvp/items/impl/enderman/DragonsCharge.java
package io.github.chaosdave34.kitpvp.items.impl.enderman;

import io.github.chaosdave34.kitpvp.abilities.Ability;
import io.github.chaosdave34.kitpvp.abilities.AbilityHandler;
import io.github.chaosdave34.kitpvp.items.CustomItem;
import net.kyori.adventure.text.Component;
import org.bukkit.Material;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class DragonsCharge extends CustomItem {
    public DragonsCharge() {
        super(Material.FIRE_CHARGE, "dragons_charge", false, true);
    }

    @Override
    public @NotNull Component getName() {
        return createSimpleItemName("Dragon's Charge");
    }

    @Override
    public @NotNull List<Ability> getAbilities() {
        return List.of(AbilityHandler.DRAGON_FIREBALL);
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/items/impl/enderman/DragonsCharge.kt
package io.github.chaosdave34.kitpvp.items.impl.enderman

import io.github.chaosdave34.kitpvp.abilities.Ability
import io.github.chaosdave34.kitpvp.abilities.AbilityHandler
import io.github.chaosdave34.kitpvp.items.CustomItem
import org.bukkit.Material

class DragonsCharge : CustomItem(Material.FIRE_CHARGE, "dragons_charge","Dragon's Charge", false, true) {
    override fun getAbilities(): List<Ability> = listOf(AbilityHandler.DRAGON_FIREBALL)
}